src/main/kotlin/bakalarka/elearningplatform/service/EnrollmentService.kt

```diff
@@ -0,0 +1,58 @@
+package bakalarka.elearningplatform.service
+
+import bakalarka.elearningplatform.db.EnrollmentRepository
+import bakalarka.elearningplatform.model.Enrollment
+import bakalarka.elearningplatform.request.AddEnrollmentRequest
+import bakalarka.elearningplatform.request.UpdateProgressRequest
+import org.springframework.stereotype.Service
+
+@Service
+class EnrollmentService(
+    var enrollmentRepository: EnrollmentRepository,
+    var courseService: CourseService
+) {
+
+    fun findByUserId(): MutableSet<Enrollment> {
+        val id = UserService.getUserId()
+        return enrollmentRepository.findByUserID(id)
+    }
+
+    fun add(request: AddEnrollmentRequest): Enrollment? {
+        val (courseId) = request
+        val userID = UserService.getUserId()
+        if (courseService.findById(courseId).isPresent) {
+            val course = courseService.findById(courseId).get()
+            if (findByUserId().any { enrollment: Enrollment -> enrollment.course == course }) {
+                return null
+            }
+            return enrollmentRepository.save(Enrollment(userID = userID, course = course))
+        } else {
+            return null
+        }
+    }
+
+    fun get(id: Long) = enrollmentRepository.findById(id)
+
+    fun findByCourseId(id: Long): Enrollment? {
+        val enrollments = findByUserId()
+        return enrollments.find { enrollment -> enrollment.course.id == id }
+    }
+
+    fun update(enrollment: Enrollment) = enrollmentRepository.save(enrollment)
+
+    fun changeChapterProgress(courseId: Long, chapterId: Long, request: UpdateProgressRequest): Enrollment? {
+        val (value) = request
+        val enrollment = findByCourseId(courseId)
+
+        return if (enrollment?.course != null) {
```

Comment:
Hmm, ak tomu spravne chapem, tak progress jedneho usera na kurz ide do jedneho riadku? Aaakoze asi je to ok, ale viac future proof by bolo, keby aj kazda chapter ide do separatneho riadku. Aaale to uz asi nechajme tak.

Reply:
Ano, progress je ako set s idckami, spring si vytvori separatnu tabulku na to

Reply:
To s tou separatnou tabulkou som nevedel. Nice, moze byt teda :+1: